### DISCUSSION TITLE
Table of contents (TOC) not generating anchor hyperlinks for <h2> tags, only ##-style headings

### DISCUSSION THREAD

**Discussion by abevoelker** [mmistakes/minimal-mistakes#2751] [Q&A] [answered]
## Environment

- Minimal Mistakes version: 4.21.0
- Ruby gem or remote theme version: Ruby gem minimal-mistakes-jekyll version 4.21.0
- Jekyll version: 3.8.7
- Git repository URL: https://github.com/mmistakes/minimal-mistakes
- GitHub Pages hosted (if yes provide URL to site): https://mmistakes.github.io/minimal-mistakes/docs/helpers
- Operating system: Ubuntu 18.04

## Expected behavior

I expect setting `toc: true` in the page front matter and then including HTML header markup in my .md files like

```markdown
<h2>Foo</h2>
```

to generate working anchor tags in the table of content (toc) sidebar, the same way it works for using markdown shorthand like

```markdown
## Foo
```

## Steps to reproduce the behavior

If you visit https://mmistakes.github.io/minimal-mistakes/docs/helpers/ right now you can see this happening in the toc there; the Placeholder 1,  Placeholder 2, Placeholder 3 text in the toc are not hyperlinks but just plain text. Here is a screenshot:

![Screenshot from 2020-12-09 23-27-36](https://user-images.githubusercontent.com/153459/101726052-67af6800-3a77-11eb-9b06-03f80e858a11.png)

I can reproduce the behavior on my own Jekyll site by writing headers like `<h2>Placeholder 1</h2>`. Writing `## Placeholder 1` instead generates the anchor link successfully.

## Other

N/A

**Comment by iBug** [comment, marked as answer]
I would rather say this is the correct behavior. The ToC template expects headings to have an HTML `id` attribute, which Jekyll's default Markdown renderer, Kramdown, will automatically generate [with `auto_ids` enabled][1].

If you want manually-written HTML headings to have ToC entries, create your `id` as well:

```html
<h2 id="placeholder-1">Placeholder 1</h2>
```

  [1]: https://github.com/mmistakes/minimal-mistakes/blob/e61476d0a9a004a0b7f2a7722481fa5589be9044/_config.yml#L194

**Reply by abevoelker** [reply]
Sounds good, I can work with that, thank you.

I'm not sure what was different with the theme in April but the documentation did not have this issue then: https://web.archive.org/web/20200415005555/https://mmistakes.github.io/minimal-mistakes/docs/helpers/

It seems the gem version at that time would've been 4.19.1, but checking out that version locally and generating the docs still has the behavior I noticed in the current version.

Anyway my issue is resolved so thanks again.

**Reply by iBug** [reply]
Hi, please check out #2752 for my analysis of the current behavior and the old (April 2020) behavior. This will be fixed in the next release of the theme.

**Reply by abevoelker** [reply]
Looks great! Thanks for your help and thoroughness. :tada:
